crazyflie/src/crazyflie_server.cpp

```diff
@@ -437,6 +543,70 @@ class CrazyflieROS
     }
   }
 
+  void on_logging_pose(uint32_t time_in_ms, const logPose* data) {
+    if (publisher_pose_) {
+      geometry_msgs::msg::PoseStamped msg;
+      msg.header.stamp = node_->get_clock()->now();
+      msg.header.frame_id = name_;
+
+      msg.pose.position.x = data->x;
+      msg.pose.position.y = data->y;
+      msg.pose.position.z = data->z;
+
+      float q[4];
+      quatdecompress(data->quatCompressed, q);
+      msg.pose.orientation.x = q[0];
+      msg.pose.orientation.y = q[1];
+      msg.pose.orientation.z = q[2];
+      msg.pose.orientation.w = q[3];
+
+      publisher_pose_->publish(msg);
```

Comment:
Could we also send transforms here like the crazyflie cflib server: https://github.com/IMRCLab/crazyswarm2/blob/119fee4414a5f3487aa4599a08ea587ecef08d11/crazyflie/scripts/crazyflie_server.py#L471-L482

Then, once RVIZ opens, you'll see the positions immediately and you don't have to set up all the poses of each drone individually for rviz. Also it will be closer to what the simulator is outputting

Reply:
I like that. The main issue is that if you fly in the mocap space, you also get the same transforms (from the mocap). The cleanest option is to use another frame_id. Perhaps pose should be in "odom", and mocap in "world"?
Another option would be to use another child_frame_id (e.g., cf_name + "_pose").

Reply:
The odom->cf or world->cf is more convention I think, but I'm not exactly sure what will happen if both are displayed... And also I'm not sure that it is still called odometry if a positioning system is used, as that usually is used for either an onboard motion sensor or wheel odometry. Technically it should be world->odom->CF where GT is providing the world ->odom transform, but in order to do that we need to have a constant stream of the onboard localization to substract that from the mocap one, which is not a given with multiple

Perhaps the last option would be best and just do world->cf for both for now, but would it be okay to do it the other way around? Call the ones from mocap positioning cf_gt (ground truth) or cf_mc (mocap) or something?

Reply:
I pushed an additional to mimic the cflib backend. I think in a future PR, I will change the default "base" frame for the motion capture (e.g., to mocap), then we don't have issues here.

If the change is fine with you, perhaps you can approve the PR @knmcguire :-)